src/js.rs

```diff
@@ -116,4 +118,120 @@ impl Backend {
 
 		Ok(None)
 	}
+
+	#[instrument(skip_all)]
+	pub async fn js_completions(
+		&self,
+		params: CompletionParams,
+		_ast: Tree,
+		rope: RopeSlice<'_>,
+	) -> anyhow::Result<Option<CompletionResponse>> {
+		let uri = &params.text_document_position.text_document.uri;
+		let position = params.text_document_position.position;
+		let Ok(ByteOffset(_offset)) = rope_conv(position, rope) else {
+			return Err(errloc!("could not find offset for {}", uri.path().as_str()));
+		};
+
+		// Get the current line
+		let line_start = rope.line_to_char(position.line as usize);
+		let line_end = if position.line as usize + 1 < rope.len_lines() {
+			rope.line_to_char(position.line as usize + 1)
+		} else {
+			rope.len_chars()
+		};
+		let line = rope.slice(line_start..line_end).to_string();
+
+		// Check if this line contains orm.call
+		if let Some(call_pos) = line.find("orm.call(") {
```

Comment:
It might be better to use a proper `query!` here instead of slicing by string; it would also help with implementing references and other helpers as well